lib/make-request.js

```diff
@@ -73,6 +73,13 @@ const makeRequest = (instance, { request: requestPhaseList, response: responsePh
       // save reference to the original callback
       const callback = _.get(ctx, 'options.callback', _.noop);
       if (requestPhasesError) {
+        // check if we got a response from caching plugin
+        if (requestPhasesError.response && requestPhasesError.response.fromCache) {
+          debug('Cached data found with response: %o and body: %o', requestPhasesError.response, requestPhasesError.body); // eslint-disable-line max-len
+          callback(null, requestPhasesError.response, requestPhasesError.body);
```

Comment:
what happens to the [Promise](https://github.com/futin/oniyi-http-client/blob/82cca1488f98352f87927bf5adb3720bda1488b2/lib/make-request.js#L71)

Reply:
struggling with "do we need to invoke the response phase list here"? Thoughts?

Reply:
To be honest, I don't like the idea of returning both promise and callback every single time. I would suggest something smarter, where if caller has provided a `callback` function, we assume that he is not interested in `request` stream object. I don't want to handle [both callback / promise errors](https://github.com/benkroeger/oniyi-http-client/blob/master/lib/make-request.test.js#L108-L115).

Or maybe even provide a flag via `options` that indicates if caller wishes to handle both callback / promise responses, or only one of them.

Reply:
As for invoking response phase list after acquiring `cached` data... I think it is not necessary. Why?
The whole purpose of using http client is to get a response. Once we have that `cached` response, it's game over. 

Would it make more sense to invoke the `caching` plugin handlers in the `final` phase(both in request / response phase lists)? This way, we can be sure the cached response went through all the previous `phases`, and that we have retrieved(in the request phase list) or cached(in the response phase list) response after all of its (potential) updates.

I think that you've already mentioned (maybe I am wrong) that `caching` plugin could be a part of `oniyi-http-client`. It would make sense to place it in the `final` phase implementation for e.x. Then, IMO there would be no need to invoke a response phase list for sure after received `cached` data.

Reply:
since each phase can have multiple handlers, you can not control that there is no handler registered after the last handlers from the cache plugin.
There are plenty of use-cases for plugins that might want to deal with data in the response phase (even if they're received from cache). One example would be counting the number of impressions of e.g. a person / user details --> how often has anyone looked at articles written by xxx?

the purpose of use the http client is to make a http request. that involves all the steps!

Reply:
also, currently the `Promise` represents the state of initiating the request, while `callback` handles the result of making a request. So IMHO we need both to be handled.

Reply:
I've created a new PR for skipping logic. 

re Promise / Callback: ok, we'll handle both callback and promise response as we did so far. 

re invoking response phase list after receiving a cached response: This requires a lot more thinking and changes on caching plugin. My suggestion is to create a separate JIRA ticket for this and work on it in the future, but for we should return cached response and not invoke response phase list. What do you think about that?